libraries/botbuilder-core/src/middlewareSet.ts

```diff
@@ -77,6 +78,15 @@ export class MiddlewareSet implements Middleware {
         this.use(...middlewares);
     }
 
+    [Symbol.iterator]() {
+        return {
+            next(): IteratorResult<MiddlewareHandler> {
+                this.count++;
```

Comment:
If we were to keep this iterator, you would actually want to do:

```typescript
[Symbol.iterator]() {
  const count = 0;
  return {
    return { done: count >= this.middleware.length, value: this.middleware[count++] };
  }
}
```

See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols#the_iterator_protocol